cmake/Platform.cmake

```diff
@@ -27,4 +30,11 @@ if (MSVC)
     # Suppress warnings: "Prefer enum class over enum" (Enum.3)
 
     add_compile_options(/wd4267 /wd4819 /wd26812)
+
+    # prevent warnings when using /std:c++17 and above
+    if(CMAKE_CXX_STANDARD GREATER 14)
```

Comment:
Very minor point, but you can use `GREATER_EQUAL 17` to make it a bit more clear that it's for 17 and above.